### DISCUSSION TITLE
Extremely poor performance in App router

### DISCUSSION THREAD

**Discussion by ArifAA14** [vercel/next.js#64301] [App Router]
Hey everyone,

I have been using NextJS for a couple of years now and I have absolutely loved it and its been my go to tool for building any kind of web apps, however since the switch to the App router the entire experience has been ruined for me. I'm struggling to understand if this is something everyone is struggling with or am I doing something wrong?

So far my experience (NextJS - **14.1.4**)

- Extremely poor load up times, every time I boot up my dev server, it takes about 20-30s which is absolutely ridiculous considering the application is nothing but the blank NextJS starter. I have never faced this issue in all my years of using the pages router.

- Hot reload isn't so hot anymore, it takes a good 4-5s I'd say?

I did some research on this and discovered that Windows Defender may cause some issues and slows down everything, so I even went ahead and disabled that but haven't noticed any differences at all. 

I did not report this as a bug cause I have seen many bug reports regarding this same issue already, I'm just trying to understand if this is a problem for everyone using the app router or maybe just an environment problem for me and how is everyone dealing with these productivity issues? 

Thanks

**Comment by icyJoseph** [comment]
In most projects I work on, this has been the reverse experience. Pages Router app, took near a minute to compile, and load, whereas the App Router app is just flying by... 

Unfortunately, without a repository, `next info` print outs, or actual screen recordings, this is very much anecdotal evidence, isn't it? 

I remember the first app I migrated to App Router, I had moved the GSSP calls to the top of my page, and that would really make hot-reloading sluggish, but as I started to break things apart, and leverage some of the caching, things just started to work well.  

Got a repo to look at?
